Design.md

```diff
@@ -66,6 +66,14 @@ Investigate the use of Option types for error returns.
 
 By making the library header only, it becomes inured against variations in compiler flags. Whether it's a static or dynamic library becomes irrelevant, and there won't be impedance mismatches.
 
+### Naming
+
+Names should follow the existing naming guidelines here: [D Style Guide](https://dlang.org/dstyle.html)
+
+When selecting a name for a type or method, a quick survey of how other popular languages name the equivalent type/method should be performed. For example, in .NET and Java, the `currTime()` method would be named `now()`. Using the same names as popular languages reduces the friction experienced by the engineer when migrating to D. Be prepared to provide examples from your survey in the Pull Request.
+
+Prefer whole words over abbreviations and dropped letters. For example, prefer `writeLine` over `writeln`. Choose the shortest name that accurately describes the feature. Abbreviations are acceptable where the abbreviation is in common usage and/or would result in a cumbersome name.
```

Comment:
I'm not sure about this one and it even counters the paragraph preceding it since this name comes from Pascal.

Reply:
Most programmers today probably don't get to see Pascal during their qualifications. Usage of it is a lot rarer than it was 30 years ago.

So it would make sense that perhaps in such cases an alias would be desirable instead of picking one over another.

Reply:
My only objection to `writeLine` is the line length. The sooner a line has to wrap, the more annoying it is, and the worse it looks on the screen. I like to try to keep line length under 80 columns so I can have more text windows on the screen. Besides, `writeln` is so commonly used it's hard to not get very used to it.

Another issue with names is their greppability. A name that matches a common word is going to have a blizzard of false matches. `writeln` is going to get pretty well-targeted hits, as opposed to, say, `write` which is not going to get very useful hits. `writeLine` is, of course, also going to do well by this metric.

Reply:
Well, first off, the 80-char limit died with the advent of widescreen monitors in the mid-2000s, and it hasn't been a thing in corporate for a couple of decades now. Everybody is on 120, and I've seen 160 limits.

Reply:
@LightBender I think your perception has biased by the kinds of projects you have worked on, which does not necessarily generalize to all fields. At my last two workplaces we've worked on multiple projects in each of the following languages: Rust, Nim, C++, Nix, Go, TypeScript (frontend and backend), and Dart (Flutter). This includes both internal projects and third-party open-source projects that we've had contribute to. Except for `go fmt`, which doesn't enforce a column limit, in my experience projects universally had a line column below 100 and often 80.

> 80-char limit died with the advent of widescreen monitors

That's completely false in my experience. I've worked on various green field projects during the past 7 years (before I was working at larger enterprise companies) and 80 columns has been the most popular choice across the different languages I mentioned earlier.

Widescreen monitors are not for reading and writing long lines (yuck), but for having more open files at the same time, without having to scroll horizontally ;)

Long lines are not a problem because of an arbitrary 80 char limit, imposed on us by dinosaurs which refuse to write code on anything but punch cards, but because they make the code more difficult to understand just like having 1000 line functions that mutate global state all over the place do. I have never found a case where a long line of code was refactored to fit in 80-100 chars (e.g. by extracting subexpressions as variables, or using LINQ/UFCS pipelines where each step of the pipeline is on a separate line) and this didn't improve the clarity.

Reply:
* Rust: https://rustc-dev-guide.rust-lang.org/conventions.html#line-length, https://doc.rust-lang.org/nightly/style-guide/#formatting-conventions
* Nim: https://nim-lang.org/docs/nep1.html#style-guidelines-spacing-and-whitespace-conventions
* https://dart.dev/effective-dart/style#avoid-lines-longer-than-80-characters
  > Readability studies show that long lines of text are harder to read because your eye has to travel farther when moving to the beginning of the next line. This is why newspapers and magazines use multiple columns of text.
   > If you really find yourself wanting lines longer than 80 characters, our experience is that your code is likely too verbose and could be a little more compact.
* Typescript: https://prettier.io/docs/en/options.html#print-width

Reply:
It's common to say 80 is silly because monitors are bigger, but I counter with: 3 way merges. If you have merge conflicts, you're going to have to stare at 3 columns of code, and even at 27" 4k it's not fun if they're significantly wider than 80.

Then there's readability. There's a reason that newspapers and magazines have columns, because too wide means less readable.

I always have 2 columns in emacs. I can deal with more than 80 in that use but not more. 160 would be unusable.

Reply:
I have routinely have VSCode open with side-by-side code tabs and the sidebar open on a 1920-wide monitor. And I use a vertical taskbar on the left side. 120/120 is easy to achieve, it's about 140 each column specifically. If I ditch the SxS mode then it's around 280 chars wide.

Columns versus rows is a usability trade-off because the fewer columns the more rows, and the less I can fit "above the fold" (to use newspaper lingo). In my experience, modern code-bases using 80 columns results in endless scrolling.

I'm happy with 120, I wasn't seriously suggesting 160 (just noting that it's quite practical on modern hardware). But 80 is an artifact of the pre-Widescreen era.

Reply:
I have my ``.editorconfig`` set to soft max 120, hard 140. Can do two files + structure view simultaneously quite comfortably.

Upping it to 100/120 at the bare minimum seems reasonable.